x/merkledb/db.go

```diff
@@ -1365,9 +1359,10 @@ func (db *merkleDB) Clear() error {
 	// Clear history
 	db.history = newTrieHistory(db.history.maxHistoryLen)
 	db.history.record(&changeSummary{
-		rootID: db.rootID,
-		values: map[Key]*change[maybe.Maybe[[]byte]]{},
-		nodes:  map[Key]*change[*node]{},
+		rootID:           db.rootID,
+		values:           map[Key]*keyChange{},
+		nodes:            map[Key]*change[*node]{},
+		sortedKeyChanges: make([]*keyChange, 0),
```

Comment:
Similar comment, but I wonder if it makes sense for us to have two copies of `keyChange` since we have to worry about them always being in sync. Maybe values could be updated to be a map of keys to indicies and do a lookup in `sortedKeyChanges`?